n3fit/src/n3fit/hyper_optimization/penalties.py

```diff
@@ -44,24 +45,25 @@ def saturation(pdf_model, stopping_object, n=100, min_x=1e-6, max_x=1e-4, flavor
     >>> from n3fit.model_gen import pdfNN_layer_generator
     >>> fake_fl = [{'fl' : i, 'largex' : [0,1], 'smallx': [1,2]} for i in ['u', 'ubar', 'd', 'dbar', 'c', 'cbar', 's', 'sbar']]
     >>> pdf_model = pdfNN_layer_generator(nodes=[8], activations=['linear'], seed=0, flav_info=fake_fl)
-    >>> isinstance(saturation(pdf_model, None), float)
+    >>> isinstance(saturation([pdf_model], None), float)
     True
 
     """
     if flavors is None:
         flavors = [1, 2]
     x = np.logspace(np.log10(min_x), np.log10(max_x), n)
     xin = np.expand_dims(x, axis=[0, -1])
-    y = pdf_model.predict([xin])
     extra_loss = 0.0
-    xpdf = y[0, :, flavors]
-    slope = np.diff(xpdf) / np.diff(np.log10(x))
-    pen = abs(np.mean(slope, axis=1)) + np.std(slope, axis=1)
-    extra_loss = np.sum(1.0 / (1e-7 + pen))
+    for pdf_model in pdf_models:
+        y = pdf_model.predict([xin])
+        xpdf = y[0, :, flavors]
+        slope = np.diff(xpdf) / np.diff(np.log10(x))
+        pen = abs(np.mean(slope, axis=1)) + np.std(slope, axis=1)
+        extra_loss += np.sum(1.0 / (1e-7 + pen))
     return extra_loss
 
 
-def patience(pdf_model, stopping_object, alpha=1e-4):
+def patience(pdf_models, stopping_object, alpha=1e-4):
```

Comment:
Could you do the same here as you did in rewards.py? Namely, using kwargs to stop pylint complaining about unused  arguments.